src/main/java/com/exadel/discount/repository/CategoryRepository.java

```diff
@@ -13,4 +19,41 @@
 
     @EntityGraph(attributePaths = {"discounts"})
     Optional<Category> findById(UUID uuid);
+
+    @Query("SELECT new com.exadel.discount.model.dto.statistics.CategoryStatisticsDTO " +
+            "(c.id, c.name, COUNT(d) AS discountsNumber, " +
+            "CASE " +
+            "   WHEN(COUNT(d) > 0) THEN SUM(d.viewNumber) " +
+            "   ELSE 0 " +
+            "END AS viewNumber, " +
+            "COUNT(cou) AS getPromoNumber) " +
+            "FROM Category c " +
+            "LEFT JOIN c.discounts d " +
+            "LEFT JOIN d.coupons cou " +
+            "ON cou.date <= :dateTo AND cou.date >= :dateFrom " +
+            "GROUP BY c ")
+    List<CategoryStatisticsDTO> getCategoriesStatistics(@Param("dateFrom") LocalDateTime dateFrom,
+                                                        @Param("dateTo") LocalDateTime dateTo,
+                                                        Pageable pageable);
```

Comment:
Do U need here Pageable?

Reply:
FE team said that we need pagination. It can return List as well as Page here